### PR TITLE
CSV separator and encoding options

### PR DESCRIPTION
**Pull request by johncmerfeld**
Addressing #20 

I'm still testing this on various separators, but for a pipe delimited file it works fine, and the CLI stuff appears to handle everything correctly.

Still need to try this on different encodings.

**One weird thing to try:** Run `scroll datascroller/demo_data/titanic_pipe.csv -s ','`. Notice that we're attempting to read a pipe-delimited file with comma separator. It prints this:
```
R0C0

        Empty DataFrame
        Columns: []
        Index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
 21, 22, 23]
```
Which is… fine but not great. I tried putting a check in for an empty dataframe, but it doesn't work because when you do that, it says the dataframe actually has 1 column and is not empty. So I'm not totally sure what that's about.

We can remove the `titanic_pipe` file if we want, but it is useful for testing.

### CODE REVIEW

**Comment by kjmerf** [comment]
Looks good overall... That is weird behavior indeed. For that particular command, I'd expect an error, which is what you get if you try:

`pd.read_csv('titanic_pipe.csv', sep=',')
`

Any idea why we're diverging from that?

I think we should keep the pipe file. When we have more sophisticated tests in place, we can run them through both file types.

I have one other nitpick. I'm used to seeing:

`program [options] required_argument`

But noticed our sequence is:

`program required_argument [options]`

I might be in the minority here but that strikes me as unconventional. Thoughts?

**Comment by johncmerfeld** [comment]
So, this needs some more work because I'm getting weird behavior.

@kjmerf about the order of arguments, this appears to be the default behavior of argparse. The order in which the user actually provides the arguments doesn't matter. They can come before or after. However, we can modify the [usage](https://docs.python.org/2/library/argparse.html#usage) parameter to make our own if we want to.

File: datascroller/command_line.py [hunk 1]
```diff
@@ -45,8 +45,14 @@ def run_scroll():
...
             # Exiting
               - q
-        """, formatter_class=argparse.RawTextHelpFormatter)
+        """
+        , usage='scroll -h csv_filepath [-s separator] [-e encoding]'
+        , formatter_class=argparse.RawTextHelpFormatter)
     parser.add_argument('csv_filepath',
                         help='a csv filepath, relative or absolute')
+    parser.add_argument("-s", "--sep", choices=[',', '|', ';', ':', ' ', '\t'], default=',""',
+                        help="csv separator, choose between ',', '|', ';', ':', ' ', '\\t'")
```
Comment (baogorek) [inline datascroller/command_line.py:54]: In the help menu, this is displaying as 
```
 -s {,,|,;,:, ,        }, --sep {,,|,;,:, ,    }
```
Maybe more escapes?

File: datascroller/command_line.py [hunk 2]
```diff
@@ -45,8 +45,14 @@ def run_scroll():
               - Ctrl + B: Page up (not working as well for some reason)
             # Exiting
               - q
-        """, formatter_class=argparse.RawTextHelpFormatter)
+        """
+        , usage='scroll -h csv_filepath [-s separator] [-e encoding]'
+        , formatter_class=argparse.RawTextHelpFormatter)
     parser.add_argument('csv_filepath',
                         help='a csv filepath, relative or absolute')
+    parser.add_argument("-s", "--sep", choices=[',', '|', ';', ':', ' ', '\t'], default=',""',
```
Comment (baogorek) [inline datascroller/command_line.py:53]: Assuming csv is the natural default, I'm having to explicitly pass in `-s ','` in order to scroll titanic.csv.

File: datascroller/scroller.py [hunk 3]
```diff
@@ -344,9 +344,13 @@ def scroll(scrollable):
...
 
-def scroll_csv(csv_path):
-    pandas_df = pd.read_csv(csv_path, dtype=object)
-    scroll(pandas_df)
+def scroll_csv(csv_path, sep=',', encoding='utf-8'):
+    pandas_df = pd.read_csv(csv_path, sep=sep, encoding=encoding, dtype=object)
+    if not pandas_df.empty:
+        scroll(pandas_df)
+        #print(pandas_df)
+    else: # TODO(johncmerfeld): Is this a bad error message?
```
Comment (baogorek) [inline datascroller/scroller.py:352]: I'd recommend raising a value error

```
else:
    raise ValueError('Empty dataframe...')
```

**Review by baogorek — COMMENTED** [review]
Still some work to do but the base functionality is there for different delimiters.

**Comment by baogorek** [comment]
Left a few comments on the code. I actually get something different when I run
 ```scroll datascroller/demo_data/titanic_pipe.csv -s ','.``` which is the whole dataframe in one column (but still many rows):

image

I think that's what I would expect. As noted in the review, I'd like to get a reasonable default working. It might be worth checking out [csv.Sniffer](https://docs.python.org/2/library/csv.html#csv.Sniffer) which might do the work for us!

**Comment by kjmerf** [comment]
@johncmerfeld per our conversation I took another crack at this one and opened a new PR

**Pull request closed without merging**